tests/test_apply.py

```diff
@@ -183,132 +161,134 @@ def test_invalid_qubit(self, init_state, device):
-        dev._obs_queue = []
-        dev.pre_measure()
+        dev.apply([qml.QubitStateVector(state, wires=wires), applied_operation])
+        dev._samples = dev.generate_samples()
 
-        res = np.fromiter(dev.probability().values(), dtype=np.float64)
-        expected = np.abs(mat @ state) ** 2
+        res = np.fromiter(dev.probability(), dtype=np.float64)
+        expected = np.abs(applied_operation.matrix @ state) ** 2
         assert np.allclose(res, expected, **tol)
 
     @pytest.mark.parametrize("theta", [0.5432, -0.232])
-    @pytest.mark.parametrize("name,func", single_qubit_param)
-    def test_single_qubit_parameters(self, init_state, device, name, func, theta, tol):
+    @pytest.mark.parametrize("operation", single_qubit_operations)
+    def test_single_qubit_operations_parameters(self, init_state, hardware_simulator_device, operation, theta, tol):
         """Test PauliX application"""
-        dev = device(1)
+        dev = hardware_simulator_device(1)
         state = init_state(1)
+        wires = [0]
+        applied_operation = operation(theta, wires=wires)
 
-        dev.apply("QubitStateVector", [0], [state])
-        dev.apply(name, [0], [theta])
-        dev._obs_queue = []
-        dev.pre_measure()
+        dev.apply([qml.QubitStateVector(state, wires=wires), applied_operation])
+        dev._samples = dev.generate_samples()
 
-        res = np.fromiter(dev.probability().values(), dtype=np.float64)
-        expected = np.abs(func(theta) @ state) ** 2
+        res = np.fromiter(dev.marginal_prob(np.abs(state) ** 2, wires), dtype=np.float64)
+        expected = np.abs(applied_operation.matrix @ state) ** 2
         assert np.allclose(res, expected, **tol)
 
-    @pytest.mark.parametrize("name,mat", two_qubit)
-    def test_two_qubit_no_parameters(self, init_state, device, name, mat, tol):
+    @pytest.mark.parametrize("operation", two_qubit)
+    def test_two_qubit_no_parameters(self, init_state, hardware_simulator_device, operation, tol):
         """Test PauliX application"""
-        dev = device(2)
+        dev = hardware_simulator_device(2)
         state = init_state(2)
+        wires = [0, 1]
 
-        dev.apply("QubitStateVector", [0, 1], [state])
-        dev.apply(name, [0, 1], [])
-        dev._obs_queue = []
-        dev.pre_measure()
+        applied_operation = operation(wires=wires)
 
-        res = np.fromiter(dev.probability().values(), dtype=np.float64)
-        expected = np.abs(mat @ state) ** 2
+        dev.apply([qml.QubitStateVector(state, wires=wires), applied_operation])
+        dev._samples = dev.generate_samples()
+
+        res = np.fromiter(dev.probability(), dtype=np.float64)
+        expected = np.abs(applied_operation.matrix @ state) ** 2
         assert np.allclose(res, expected, **tol)
 
     @pytest.mark.parametrize("mat", [U, U2])
-    def test_qubit_unitary(self, init_state, device, mat, tol):
+    def test_qubit_unitary(self, init_state, hardware_simulator_device, mat, tol):
```

Comment:
How come these tests don't work on statevector simulator backends any more (like they used to?)

Reply:
Thanks for this question! I just realized that I've had a confusion here: the first two test classes checked the cases for `analytic=True` whereas the second `analytic=False` for **all backends**. The reason for this change was that I thought that they were completely based on backends.

I've reverted to using the `device` fixture everywhere and updated the class names and docstrings.